src/libcharon/plugins/kernel_super/picomsg_conversions.c

```diff
@@ -65,6 +65,21 @@ void ts_to_prefix_string_for_logs(struct traffic_selector_t* ts,
 	pm_proto_prefix_to_string_for_logs(&route_prefix, str);
 }
 
+bool ts_to_prefix_string(struct traffic_selector_t* ts,
+						char str[MAX_IP_PREFIX_STR_LEN]) {
+	struct pm_proto_prefix prefix;
```

Comment:
Now this function and `ts_to_prefix_string_for_logs()` are now basically identical.
Make sure not to have repeating logics if can help it. Instead make `ts_to_prefix_string_for_logs()` call this one:

``` cpp
void ts_to_prefix_string_for_logs(struct traffic_selector_t* ts,
								  char str[MAX_IP_PREFIX_STR_LEN]) {
    if (! ts_to_prefix_string(ts, str)) {
		safe_strncpy(str, "(invalid)", MAX_IP_PREFIX_STR_LEN);
		return false;
    }
    return true;
}
```